core/contact/service.go

```diff
@@ -0,0 +1,169 @@
+package contact
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+
+	"github.com/anyproto/any-sync/app"
+	"github.com/anyproto/any-sync/util/crypto"
+
+	"github.com/anyproto/anytype-heart/core/block/editor/userdataobject"
+	"github.com/anyproto/anytype-heart/pkg/lib/logging"
+	"github.com/anyproto/anytype-heart/pkg/lib/pb/model"
+	"github.com/anyproto/anytype-heart/space"
+)
+
+var log = logging.Logger(CName)
+
+const CName = "core.contact.service"
+
+type Service interface {
+	app.ComponentRunnable
+	SaveContact(ctx context.Context, identity string, profileSymKey string) error
+	DeleteContact(ctx context.Context, identity string) error
+}
+
+type identityService interface {
+	app.Component
+	RegisterIdentity(spaceId string, identity string, encryptionKey crypto.SymKey, observer func(identity string, profile *model.IdentityProfile)) error
+	UnregisterIdentity(spaceId string, identity string)
+	WaitProfile(ctx context.Context, identity string) *model.IdentityProfile
+	GetIdentityKey(identity string) crypto.SymKey
+}
+
+type service struct {
+	identityService identityService
+	ctx             context.Context
+	cancel          context.CancelFunc
+	spaceService    space.Service
+}
+
+func (s *service) Run(ctx context.Context) (err error) {
+	s.ctx, s.cancel = context.WithCancel(context.Background())
+	return s.registerExistingContacts(ctx)
+}
+
+func (s *service) registerExistingContacts(ctx context.Context) error {
```

Comment:
In a gist this is the service that should be responsible for managing the contact objects (i.e. smart blocks):
- update them
- create them
- delete them (remove from indexes and from cache)